text-minimessage/src/main/java/net/kyori/adventure/text/minimessage/serializer/package-info.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * This file is part of adventure, licensed under the MIT License.
+ *
+ * Copyright (c) 2017-2022 KyoriPowered
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+/**
+ * Serializer for the MiniMessage language.
+ *
+ * <p>This is currently internal API, likely to change at any time. It may or may not ever become stable.</p>
+ *
+ * @since 4.10.0
+ */
+@org.jetbrains.annotations.ApiStatus.Internal
```

Comment:
I'm not sure that coveys how much others should not use this API. There are a few things that bug me about this API that probably won't be changed before merge, so it will definitely be breaking at some point in the future.

Plus, the vast majority of external users won't have any way to implement their own serializers at this point -- the set of things that can be serialized is somewhat fixed based on the Vanilla component types and style attributes. Custom tags would be difficult to reliably rediscover based on the component tree.